WordPress/src/main/java/org/wordpress/android/ui/FilteredRecyclerView.java

```diff
@@ -231,15 +274,100 @@ public void hideLoadingProgress() {
         }
     }
 
+    /*
+     * use this to add custom control/component views to your toolbar. Components are added on the right side of the
+      * toolbar */
+    public void addToolbarCustomControl(View v, OnClickListener clickListener){
+        if (v != null){
+            mCustomComponentsContainer.addView(v);
+            if (clickListener != null){
+                v.setOnClickListener(clickListener);
+            }
+        }
+    }
+
+    public void setToolbarBackgroundColor(int color){
+        mToolbar.setBackgroundColor(color);
+    }
+
+    public void setToolbarSpinnerTextColor(int color){
+        mSpinnerTextColor = color;
+    }
+
+    public void setToolbarSpinnerDrawable(int drawableResId){
+        mSpinnerDrawableRight = drawableResId;
+    }
+
+    public void setToolbarLeftPadding(int paddingLeft){
+        mToolbar.setPadding(paddingLeft,
+                mToolbar.getPaddingTop(),
+                mToolbar.getPaddingRight(),
+                mToolbar.getPaddingBottom());
+    }
+
+    public void setToolbarRightPadding(int paddingRight){
+        mToolbar.setPadding(
+                mToolbar.getPaddingLeft(),
+                mToolbar.getPaddingTop(),
+                paddingRight,
+                mToolbar.getPaddingBottom());
+    }
+
+    public void setToolbarLeftAndRightPadding(int paddingLeft, int paddingRight){
+        mToolbar.setPadding(
+                paddingLeft,
+                mToolbar.getPaddingTop(),
+                paddingRight,
+                mToolbar.getPaddingBottom());
+    }
+
+    public void scrollRecycleViewToPosition(int position) {
+        if (mRecyclerView == null) return;
+
+        mRecyclerView.scrollToPosition(position);
+    }
+
+    public int getCurrentPosition() {
+        if (mRecyclerView != null && mRecyclerView.getLayoutManager() != null) {
+            return ((LinearLayoutManager) mRecyclerView.getLayoutManager()).findFirstVisibleItemPosition();
+        } else {
+            return -1;
+        }
+    }
+
+    public void smoothScrollToPosition(int position){
+        if (mRecyclerView != null && mRecyclerView.getLayoutManager() != null) {
+            mRecyclerView.getLayoutManager().smoothScrollToPosition(mRecyclerView, null, position);
+        }
+    }
+
+    /*
+    * use this to access the internal RecyclerView if you need to, for example, add items decoration
+    * */
+    public RecyclerView getInternalRecyclerView(){
```

Comment:
I should preface this by saying I really don't have strong feelings either way - it's simply an interesting conversation to have.

I definitely understand the desire to expose the internal recycler for possible future use, but I lean towards waiting for those future uses to exist first (I say "lean towards" because I'm sure I've violated that rule countless times in my own code).

In this particular case, it seems inconsistent to expose the internal recycler for item decorations, yet still have methods like `scrollRecycleViewToPosition()`, `smoothScrollToPosition()` and `getCurrentPosition()`, all of which operate on the internal recycler.

But perhaps the most important reason to avoid this is so callers don't have to bother checking whether the internal recycler is null. The three routines mentioned above all check for null, freeing the caller from the responsibility to do so.

Have I mentioned that I hate having to null check all the time in Java? :)